swift/extractor/infra/SwiftMangledName.h

```diff
@@ -25,28 +24,40 @@ struct SwiftMangledName {
   SwiftMangledName(SwiftMangledName&&) = default;
   SwiftMangledName& operator=(SwiftMangledName&&) = default;
 
+  template <typename... Args>
+  SwiftMangledName(Args&&... args) {
+    (operator<<(std::forward<Args>(args)), ...);
+  }
+
   // streaming labels or ints into a SwiftMangledName just appends them
   SwiftMangledName& operator<<(UntypedTrapLabel label) &;
   SwiftMangledName& operator<<(unsigned i) &;
 
+  template <typename Tag>
+  SwiftMangledName& operator<<(TrapLabel<Tag> label) & {
+    return operator<<(static_cast<UntypedTrapLabel>(label));
+  }
+
   // streaming string-like stuff will add a new part it only if strictly required, otherwise it will
   // append to the last part if it is a string
   template <typename T>
   SwiftMangledName& operator<<(T&& arg) & {
-    if (parts.empty() || !std::holds_alternative<std::string>(parts.back())) {
-      parts.emplace_back("");
-    }
-    std::get<std::string>(parts.back()) += std::forward<T>(arg);
+    value += arg;
     return *this;
   }
 
-  SwiftMangledName& operator<<(SwiftMangledName&& other) &;
-  SwiftMangledName& operator<<(const SwiftMangledName& other) &;
+  SwiftMangledName& operator<<(const SwiftMangledName& other) {
```

Comment:
Optional: If we change the named `str()` to `operator std::string`, we can get rid of this overload.

Reply:
good idea! `operator std::string_view` works even better (no copy)

Reply:
I'm doing this in https://github.com/github/codeql/pull/12653 as it turns out we require that for the whole thing to actually work.

Reply:
actually, changed my mind, I will just push https://github.com/github/codeql/pull/12653 on top of this, you can review the two new commits